marketplace-and-integrations/packages/authorized-services.md

```diff
@@ -0,0 +1,776 @@
+
+###### IdentityHost *
+
+The host name for the service's authentication endpoint, used to initiate the authorization of the service by asking the user to login. For GitHub, this is `https://github.com`.
+
+###### TokenHost
+
+Some providers make available a separately hosted service for handling requests for access tokens. If that's the case, it can be provided here. If not provided, the value of `IdentityHost` is used. For GitHub, this is not necessary as the value is `https://github.com`, the same as the identity host.
+
+###### RequestIdentityPath *
+
+Used, along with `IdentityHost` to construct a URL that the user is redirected to when initiating the authorization of the service via the backoffice. For GitHub, the required value is `/login/oauth/authorize`.
+
+###### AuthorizationUrlRequiresRedirectUrl
+
+Some providers require a redirect URL to be provided with the authentication request. For others, instead it's necessary to configure this as part of the registered app. The default value if not provided via configuration is `false`, which is sufficient for the GitHub example.
+
+###### RequestTokenPath *
+
+Used, along with `TokenHost` to construct a URL used for retrieving access tokens. For GitHub, the required value is `/login/oauth/access_token`.
+
+###### RequestTokenFormat
+
+An enum value that controls how the request to retrieve an access token is formatted. Options are `Querystring` and `FormUrlEncoded`. `Querystring` is the default value and is used for GitHub.
+
+###### JsonSerializer
+
+An enum value that defines the JSON serializer to use when creating requests and deserializing responses. Options are `Default` and `JsonNet` and `SystemTextJson`.
+
+- `Default` - uses the Umbraco CMS default `IJsonSerializer`.
+- `JsonNet` - uses the JSON.Net serializer.
+- `SystemTextJson` - uses the System.Text.Json serializer.
+
+###### AuthorizationRequestRequiresAuthorizationHeaderWithBasicToken
+
+This flag indicates whether the basic token should be included in the request for access token. If true, a base64 encoding of <clientId>:<clientSecret> will be added to
+the authorization header.
+
+###### ClientId *
+
+This value will be retrieved from the registered service app.
+
+###### ClientSecret *
+
+This value will be retrieved from the registered service app.  As the name suggests, it should be kept secret and so is probably best not added directly to `appSettings.json` and checked into source control.
+
+###### UseProofKeyForCodeExchange *
+
+This flag will extend the OAuth flow with an additional security layer called [PKCE (Proof Key for Code Exchange)](https://auth0.com/docs/get-started/authentication-and-authorization-flow/authorization-code-flow-with-proof-key-for-code-exchange-pkce).
+
+In the OAuth with PKCE (Proof Key for Code Exchange) flow, a random code will be generated on the client and stored under the name `code_verifier`. Using the `SHA-256` algorithm it will be hashed under the name `code_challenge`.
```

Comment:
Acronyms should only need to be define once 🤞 

```suggestion
In the OAuth with PKCE flow, a random code will be generated on the client and stored under the name `code_verifier`. Using the `SHA-256` algorithm it will be hashed under the name `code_challenge`.
```